packages/web3-common/src/types.ts

```diff
@@ -38,6 +38,15 @@ export interface JsonRpcNotification<T = JsonRpcResult> {
 	readonly jsonrpc: JsonRpcIdentifier;
 	readonly method: string; // for subscription
 	readonly params: SubscriptionParams<T>; // for subscription results
+	readonly result: never;
+}
+
+export interface JsonRpcSubscriptionResult {
```

Comment:
this is for getting subscription id in result field?

Reply:
Subscription Id is not **in** the result field rather the result field itself. 

```
>> {"id": 1, "method": "eth_subscribe", "params": ["newHeads", {}]}
<< {"jsonrpc":"2.0","id":1,"result":"0xcd0c3e8af590364c09d0fa6a1210faf5"}
```